spec/synchronizers/bookingsync_portal/write/source_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+describe BookingsyncPortal::Write::Source do
```

Comment:
How about doing some integration tests with VCR and webmock? I think best to use some VCR cassettes and use webmock syntax to verify if the proper request has been performed or not - `assert_requested` and `assert_not_requested`, used to be a fan of mocks, but now I think they are really brittle and they focus too much on internal details.